src/Actions/RetryBulkRequest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace JustBetter\MagentoAsync\Actions;
+
+use Illuminate\Database\Eloquent\Model;
+use JustBetter\MagentoAsync\Client\MagentoAsync;
+use JustBetter\MagentoAsync\Contracts\RetriesBulkRequest;
+use JustBetter\MagentoAsync\Enums\OperationStatus;
+use JustBetter\MagentoAsync\Models\BulkOperation;
+use JustBetter\MagentoAsync\Models\BulkRequest;
+use JustBetter\MagentoClient\Client\Magento;
+
+class RetryBulkRequest implements RetriesBulkRequest
+{
+    public function __construct(protected MagentoAsync $client)
+    {
+    }
+
+    public function retry(BulkRequest $bulkRequest, bool $onlyFailed): ?BulkRequest
+    {
+        /** @var array<int, mixed> $payload */
+        $payload = [];
+        /** @var array<int, ?Model> $subjects */
+        $subjects = [];
+
+        $operations = $bulkRequest->operations;
+
+        foreach ($bulkRequest->request as $index => $request) {
+
+            /** @var BulkOperation $operation */
+            $operation = $operations->where('operation_id', '=', $index)->firstOrFail();
+
+            if ($onlyFailed && ! in_array($operation->status, OperationStatus::failedStatuses())) {
+                continue;
+            }
+
+            $payload[] = $request;
+            if ($operation->subject !== null) {
```

Comment:
You are always adding the subject, even if it's null, so this check should not be needed. Does this not cause any unwanted side effects with e.g. events?